src/index.js

```diff
@@ -0,0 +1,752 @@
+   * @see {@link https://bynder.docs.apiary.io/#reference/upload-assets/4-finalise-a-completely-uploaded-file/finalise-a-completely-uploaded-file}
+   * @param {String} fileId Unique file identifier
+   * @param {String} fileName Original file name
+   * @param {Number} chunksCount Number of chunks
+   * @param {Number} fileSize File byte size
+   * @return {Promise<object>} Object containing the correlation ID (`correlationId`) and file ID (`fileId`) of the upload
+   */
+  finaliseUpload(fileId, fileName, chunksCount, fileSize) {
+    return this.api.send('POST', `v7/file_cmds/upload/${fileId}/finalise`, {
+      chunksCount, fileName, fileSize
+    })
+      .then(response => response.headers['x-api-correlation-id']);
+  }
+
+  /**
+   * Prepares the remote env to upload a new file
+   * @return {Promise<string>} Unique file identificator
+   */
+  prepareUpload() {
+    return this.api.send('POST', 'v7/file_cmds/upload/prepare')
+      .then(response => response.file_id);
+  }
+
+  /**
+   * Saves a media asset in Bynder. If media id is specified in the data a new version of the asset will be saved.
+   * Otherwise a new asset will be saved.
+   * @see {@link https://bynder.docs.apiary.io/#reference/upload-assets/4-finalise-a-completely-uploaded-file/save-as-a-new-asset}
+   * @param {Object} data Asset data
+   * @return {Promise<object>}
+   */
+  saveAsset(data) {
+    if (!data.brandId) {
+      return rejectValidation('upload', 'brandId');
+    }
+
+    const { fileId } = data;
+    const url = fileId ? `v4/media/${fileId}/save/` : 'v4/media/save/';
+    return this.api.send('POST', url, data);
+  }
+
+  /**
+   * Uploads arbirtrarily sized buffer or stream file to provided S3 endpoint in chunks and registers each chunk to Bynder.
+   * Resolves the passed init result and final chunk number.
+   * @async
+   * @param {Object} file An object containing the id of the desired collection.
+   * @param {String} file.filename The file name of the file to be saved
+   * @param {Buffer|Readable} file.body The file to be uploaded. Can be either buffer or a read stream.
+   * @param {Number} file.length The length of the file to be uploaded
+   * @param {String} fileId Unique file identifier
+   * @param {Number} size File byte size
+   * @return {Promisee<number>} Total number of chunks uploaded to the upload payload
+   */
+  async uploadFileInChunks(file, fileId, size) {
+    const { body } = file;
+    // Developers can use this to track file upload progress
+    this._chunks = Math.ceil(size / FILE_CHUNK_SIZE);
+    this._chunkNumber = 0;
+
+    // Iterate over the chunks and send them
+    while (this._chunkNumber <= (this._chunks - 1)) {
+      const start = this._chunkNumber * FILE_CHUNK_SIZE;
+      const end = Math.min(start + FILE_CHUNK_SIZE, size);
+      const chunk = body.slice(start, end);
+
+      await this.api
+        .send('POST', `v7/file_cmds/upload/${fileId}/chunk/${this._chunkNumber}`, { chunk })
```

Comment:
Here FS is expecting a sha-256 to be passed for each chunk. They verify that on their end to check for corrupted chunks.

` headers["content-sha256"] = sha256(each_chunk).hexdigest` (This is the python version of this which)

Reply:
See steps 36-38 [here](https://api.media.atlassian.com/file/668ea053-fb36-4122-812b-046fb645cc37/binary?token=eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiI0MmU0MjMyMy0zYzcwLTRiYmEtYjJkMC1mYTQ0YmQ4NzAxNjIiLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOjY2OGVhMDUzLWZiMzYtNDEyMi04MTJiLTA0NmZiNjQ1Y2MzNyI6WyJyZWFkIl19LCJleHAiOjE2MDcwODMwODAsIm5iZiI6MTYwNzAwMDEwMH0.2Tg8bAzJaPw7dCgi65MyNvTcQX-f4c2dmgNq5WwE5Ro&client=42e42323-3c70-4bba-b2d0-fa44bd870162&name=upload-streamlined.png)